include/fiction/io/write_svg_layout.hpp

```diff
@@ -771,6 +1139,116 @@ void write_qca_layout_svg(const Lyt& lyt, const std::string_view& filename, cons
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt must be a SiDB layout");
+    static_assert(!is_sidb_defect_surface_v<Lyt>, "SiDB defects are not supported");
+
+    detail::write_sidb_layout_svg_impl<Lyt> p{lyt, os, ps};
+    p.run();
+}
+
+/**
+ * Writes an SVG representation of an SiDB cell-level SiDB layout into a file.
+ *
+ * @note SiDB defects are not supported yet.
+ *
+ * @tparam Lyt SiDB cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param filename The file name to create and write into.
+ * @param ps Parameters.
+ */
+template <typename Lyt>
+void write_sidb_layout_svg(const Lyt& lyt, const std::string_view& filename,
+                           const write_sidb_layout_svg_params& ps = {})
+{
+    static_assert(is_cell_level_layout_v<Lyt>, "Lyt is not a cell-level layout");
+    static_assert(has_sidb_technology_v<Lyt>, "Lyt must be a SiDB layout");
+    static_assert(!is_sidb_defect_surface_v<Lyt>, "SiDB defects are not supported");
+
+    std::ofstream os{filename.data(), std::ofstream::out};
+
+    if (!os.is_open())
+    {
+        throw std::ofstream::failure("Could not open file");
+    }
+
+    write_sidb_layout_svg(lyt, os, ps);
+    os.close();
+}
+
+/**
+ * Writes an SVG representation of an cell-level layout into an output stream.
+ *
+ * @tparam Lyt cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param os The output stream to write into.
+ * @param ps Parameters.
+ */
+template <typename Lyt>
+void write_svg_layout(const Lyt& lyt, std::ostream& os, const write_svg_layout_params& ps = {})
+{
+    if constexpr (has_qca_technology_v<Lyt>)
+    {
+        write_qca_layout_svg(lyt, os, ps.qca_params);
+    }
+    else if constexpr (has_sidb_technology_v<Lyt>)
+    {
+        write_sidb_layout_svg(lyt, os, ps.sidb_params);
+    }
+    else
+    {
+        throw std::invalid_argument("Unknown layout type");
+    }
+}
+
+/**
+ * Writes an SVG representation of an cell-level layout into a file.
+ *
+ * @note SiDB defects are not supported yet.
+ *
+ * @tparam Lyt Cell-level layout type.
+ * @param lyt The layout to be written.
+ * @param filename The file name to create and write into.
+ * @param ps Parameters.
+ */
+template <typename Lyt>
+void write_svg_layout(const Lyt& lyt, const std::string_view& filename, const write_svg_layout_params& ps = {})
+{
+    std::ofstream os{filename.data(), std::ofstream::out};
+
+    if (!os.is_open())
+    {
+        throw std::ofstream::failure("Could not open file");
+    }
+
+    if constexpr (has_qca_technology_v<Lyt>)
+    {
+        write_qca_layout_svg(lyt, os, ps.qca_params);
+    }
+    else if constexpr (has_sidb_technology_v<Lyt>)
+    {
+        write_sidb_layout_svg(lyt, os, ps.sidb_params);
+    }
+    else
+    {
+        throw std::invalid_argument("Unknown layout type");
+    }
+    os.close();
+}
```

Comment:
What do we need these generic functions for?

Reply:
We do not need it. I thought it would be nice since we do it similiar in the ``print_layout`` function.